### PR TITLE
fix driver.IndexFromMachineName()

### LINKED ISSUE #10820: Flake Tests: fix driver.IndexFromMachineName()
driver.IndexFromMachineName() could return unexpected results for randomly-generated node names in tests

### example
https://storage.googleapis.com/minikube-builds/logs/10683/927d8b9/Docker_Linux.html#fail_TestOffline

```
=== RUN   TestOffline
aab_offline_test.go:54: (dbg) Run:  out/minikube-linux-amd64 start -p offline-docker-20210314040449-6655 --alsologtostderr -v=1 --memory=2000 --wait=true --driver=docker  --container-runtime=docker
...
	I0314 04:04:50.291570   64873 network_create.go:86] found free network subnet for docker: &{IP:192.168.79.0 Netmask:255.255.255.0 CIDR:192.168.79.0/24 Gateway:192.168.79.1 ClientMin:192.168.79.2 ClientMax:192.168.79.254 Broadcast:192.168.79.255 Interface:{IfaceName: IfaceIPv4: IfaceMTU:0 IfaceMAC:}}
	I0314 04:04:50.291587   64873 network_create.go:105] attempt to create network 192.168.79.0/24 with subnet: offline-docker-20210314040449-6655 and gateway 192.168.79.1 and MTU of 1500 ...
	I0314 04:04:50.291662   64873 cli_runner.go:115] Run: docker network create --driver=bridge --subnet=192.168.79.0/24 --gateway=192.168.79.1 -o --ip-masq -o --icc -o com.docker.network.driver.mtu=1500 --label=created_by.minikube.sigs.k8s.io=true offline-docker-20210314040449-6655
	I0314 04:04:50.411722   64873 kic.go:102] calculated static IP "192.168.79.0" for the "offline-docker-20210314040449-6655" container
...
```
and the tests would then fail with the node's ip address of 192.168.79.0...

explanation - currently, IndexFromMachineName https://github.com/kubernetes/minikube/blob/ff0e25ada888cf1bd7b262857ce96531c1231e69/pkg/minikube/driver/driver.go#L322-L332 would return :
> "offline-docker-20210314040449-6655" => 6655

then in https://github.com/kubernetes/minikube/blob/ff0e25ada888cf1bd7b262857ce96531c1231e69/pkg/drivers/kic/kic.go#L99-L103
192.168.79.1 gateway would result in `192.168.79.0` node ip address for name offline-docker-20210314040449-`6655`

similarly, for node name offline-docker-20210314040449-`6654`, it would result in `192.168.79.255` node ip

/assign

### PR DESCRIPTION
**Pull request by prezha**
fixes #10820

as detailed in the issue's description, this pr fixes driver.IndexFromMachineName() and extends the tests to cover for noted edge cases

### CODE REVIEW

File: pkg/minikube/driver/driver.go [hunk 1]
```diff
@@ -321,12 +321,16 @@ func SetLibvirtURI(v string) {
...
+	if strings.HasPrefix(m, "m") { // likely minion node
+		m = strings.TrimPrefix(m, "m")
+		i, err := strconv.Atoi(m)
+		if err != nil {
+			return 1 // master node
+		}
+		return i // minion node
 	}
-	return i
+	return 1 // master node
```
Comment (medyagh) [inline pkg/minikube/driver/driver.go:335]: do u mind adding unit tests for this ? it is not clear what issue this is fixing and I rather we have clear idea with uni tests

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/driver/driver.go (hunk 1 above)
Comment (medyagh) [reply pkg/minikube/driver/driver.go:335]: do we hve an example failure we could see ?

Comment (prezha) [reply pkg/minikube/driver/driver.go:335]: yes, i've given examples in the issue description itself: https://github.com/kubernetes/minikube/issues/10820 and the pr also includes additional tests that should cover those

examples given in the issue were:
```
192.168.79.1 gateway would result in 192.168.79.0 node ip address for name offline-docker-20210314040449-6655
similarly, for node name offline-docker-20210314040449-6654, it would result in 192.168.79.255 node ip
```

Comment (medyagh) [reply pkg/minikube/driver/driver.go:335]: thanks for the update, would u plz add tests to TestIndexFromMachineName too ? in addition to TestIndexFromMachineNameClusterConfig ?

https://github.com/kubernetes/minikube/blob/1f75634357d1ee7ae44ed02705557353401dd331/pkg/minikube/driver/driver_test.go#L211

Comment (prezha) [reply pkg/minikube/driver/driver.go:335]: np, i've added also TestIndexFromMachineName tests

**Review by spowelljr — APPROVED** [review]

**Review by medyagh — COMMENTED** [review]
Thank you, lgtm
